util.js

```diff
@@ -62,3 +62,24 @@ exports.merge = {
     return merge(manf, _manf)
   }
 }
+
+exports.hookOptionalCB = function (syncFn) {
+  // syncFn is a function that's expected to return its result or throw an error
+  // we're going to hook it so you can optionally pass a callback
+  syncFn.hook(function(fn, args) {
+    // if a function is given as the last argument, treat it as a callback
+    var cb = args[args.length - 1]
+    if (typeof cb == 'function') {
+      try {
+        var res = fn.apply(this, args)
+        cb(null, res)
```

Comment:
This should probably pop the cb off the array, to ensure that the funciton acts like a proper sync function and can't call it twice.